README.txt
Parçala, Hj-Split ile aynı işi yapan dosya parçalama ve birleştirme yazılımıdır.

Parçala, büyük boyutlu dosyaları parçalara böldüğü gibi parçalanmış
dosyaları birleştirme ve dosyaların doğrulamasını yapabilmektedir.


Bağımlılıklar:

PyQt4






CHANGELOG.txt
===============================
Version 1.8
===============================

* Yeni ana pencere oluşturuldu.
* Doğrulama işlevi çalıştırıldı.
* Modüler yapı oluşturuldu.
* Birleştirme ve parçalama işlevi çalıştırıldı.
* Süreçler transparan siyah bir widgette işliyor.
* Birleştirme ve parçalama da dosya varlığı kontrolü eklendi.

===============================
Version 1.6
===============================

*  Bir dosyayı eşit parçaya bölme işlemi başarısız oluyordu, düzeltildi.
*  File dialog Windowsta açılmıyordu düzeltildi.

===============================
Version 1.5
===============================

*  Doğrulama işlemine sha1 de eklendi.
*  CSS kodları kaldırıldı.
*  Doğrulamada okunan verinin rame yüklenmesi düzeltildi.
*  md5 ve sha1 fonksiyonları Hash sınıfına alındı.
*  Hata mesajları düzenlendi.

===============================
Version 1.1
===============================

*  Yazılamayan yerlere yazmaya çalışma hatası düzeltildi.
*  Nereye butonuyla girdiğiniz isim değişmiyordu düzeltildi.

===============================
Version 1.0
===============================

*  Doğrulama özelliği eklendi.
*  Birleştirme özelliği eklendi.
*  Parçalama özelliği eklendi.

parcala/lib/hakkinda.py
# -*- coding:utf-8 -*-

#Copyright (C) 2011, Metehan Özbek
#
#This program is free software: you can redistribute it and/or modify
#it under the terms of the GNU General Public License as published by
#the Free Software Foundation, either version 3 of the License, or
#(at your option) any later version.
#
#This program is distributed in the hope that it will be useful,
#but WITHOUT ANY WARRANTY; without even the implied warranty of
#MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
#GNU General Public License for more details.
#
#You should have received a copy of the GNU General Public License
#along with this program. If not, see <http://www.gnu.org/licenses/>.

from PyQt4.QtGui import QDialog, QGridLayout, QLabel, QSpacerItem, QSizePolicy, QPixmap, QFont, QGroupBox, \
    QScrollArea, QFrame, QWidget, QApplication
from PyQt4.QtCore import Qt



class Hakkinda(QDialog):
    def __init__(self, parent):
        QDialog.__init__(self, parent)
        self.resize(500, 350)
        self.setMaximumSize(500, 350)
        self.gLayout = QGridLayout(self)
        self.logo = QLabel(self)
        self.logo.setPixmap(QPixmap(":/resim/parcala.png"))
        self.gLayout.addWidget(self.logo, 0, 0, 2, 1)
        self.appName = QLabel(self)
        font = QFont()
        font.setPointSize(32)
        font.setWeight(50)
        self.appName.setFont(font)
        self.gLayout.addWidget(self.appName, 0, 1, 1, 2)
        self.appVersion = QLabel(self)
        font = QFont()
        font.setPointSize(9)
        font.setWeight(75)
        font.setBold(True)
        self.appVersion.setFont(font)
        self.appVersion.setAlignment(Qt.AlignHCenter|Qt.AlignTop)
        self.gLayout.addWidget(self.appVersion, 1, 1, 1, 2)
        self.gBox = QGroupBox(self)
        font = QFont()
        font.setPointSize(12)
        font.setWeight(75)
        font.setBold(True)
        self.gBox.setFont(font)
        self.gLayout2 = QGridLayout(self.gBox)
        self.scrollArea = QScrollArea(self.gBox)
        self.scrollArea.setFrameShape(QFrame.NoFrame)
        self.scrollArea.setWidgetResizable(True)

        self.scrollAreaWidgetContents = QWidget()
        self.scrollAreaWidgetContents.setGeometry(0, 0, 476, 199)

        self.gLayout3 = QGridLayout(self.scrollAreaWidgetContents)
        self.appHakkinda = QLabel(self.scrollAreaWidgetContents)
        font = QFont()
        font.setPointSize(9)
        font.setWeight(50)
        font.setBold(False)
        self.appHakkinda.setFont(font)
        self.appHakkinda.setWordWrap(True)
        self.gLayout3.addWidget(self.appHakkinda, 0, 0, 1, 1)
        self.scrollArea.setWidget(self.scrollAreaWidgetContents)
        self.gLayout2.addWidget(self.scrollArea, 0, 0, 1, 1)
        self.gLayout.addWidget(self.gBox, 2, 0, 2, 4)
        spacerItem = QSpacerItem(40, 20, QSizePolicy.Expanding, QSizePolicy.Minimum)
        self.gLayout.addItem(spacerItem, 0, 3, 1, 1)
        spacerItem1 = QSpacerItem(20, 40, QSizePolicy.Minimum, QSizePolicy.Expanding)
        self.gLayout.addItem(spacerItem1, 2, 1, 1, 2)

        self.setWindowTitle(self.trUtf8(u"%s Hakkında"%QApplication.applicationName()))
        self.appName.setText(self.trUtf8(u"%s"%QApplication.applicationName()))
        self.appVersion.setText(self.trUtf8(u"Sürüm %s"%QApplication.applicationVersion()))
        self.gBox.setTitle(self.trUtf8("Hakkında"))
        self.appHakkinda.setText(self.trUtf8("""
        <p>Parçala, Hj-Split ile aynı işi yapan dosya parçalama ve birleştirme yazılımıdır.</p>
        <p>Parçala, büyük boyutlu dosyaları parçalara böldüğü gibi parçalanmış dosyaları birleştirme ve dosyaların doğrulamasını yapabilmektedir.</p>

        <p><b>Geliştirici:</b> Metehan Özbek - <a href='mailto:metehan@metehan.us'>metehan@metehan.us</a></p>
        <p><b>Görsel Çalışma:</b> Yasin Özcan - <a href='mailto:hamfindik@gmail.com'>hamfindik@gmail.com</a></p>
        <p><b>Lisans:</b> GPL v3</p>
        <p></p>"""))



parcala/language/tr_TR.ts
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE TS>
<TS version="2.0" language="tr_TR" sourcelanguage="tr_TR">
<context>
    <name>Parcala</name>
    <message utf8="true">
        <location filename="parcala.py" line="42"/>
        <source>Parçala</source>
        <translation>Parçala</translation>
    </message>
    <message utf8="true">
        <location filename="parcala.py" line="48"/>
        <source>Birleştir</source>
        <translation>Birleştir</translation>
    </message>
    <message utf8="true">
        <location filename="parcala.py" line="54"/>
        <source>Doğrula</source>
        <translation>Doğrula</translation>
    </message>
    <message utf8="true">
        <location filename="parcala.py" line="60"/>
        <source>Hakkında</source>
        <translation>Hakkında</translation>
    </message>
    <message>
        <location filename="parcala.py" line="65"/>
        <source>%s %s u00a9 2011 - www.metehan.us</source>
        <translation>%s %s u00a9 2011 - www.metehan.us</translation>
    </message>
</context>
<context>
    <name>KaraWidget</name>
    <message>
        <location filename="karawidget.py" line="64"/>
        <source>Hata!</source>
        <translation>Hata!</translation>
    </message>
    <message utf8="true">
        <location filename="karawidget.py" line="54"/>
        <source>Aynı isimde dosya(lar) mevcut.</source>
        <translation>Aynı isimde dosya(lar) mevcut.</translation>
    </message>
    <message utf8="true">
        <location filename="karawidget.py" line="54"/>
        <source>Üzer(ler)ine Yaz</source>
        <translation>Üzer(ler)ine Yaz</translation>
    </message>
    <message utf8="true">
        <location filename="karawidget.py" line="64"/>
        <source>İptal</source>
        <translation>İptal</translation>
    </message>
    <message utf8="true">
        <location filename="karawidget.py" line="64"/>
        <source>Aynı isimde dosya mevcut.</source>
        <translation>Aynı isimde dosya mevcut.</translation>
    </message>
    <message utf8="true">
        <location filename="karawidget.py" line="64"/>
        <source>Üzerine Yaz</source>
        <translation>Üzerine Yaz</translation>
    </message>
</context>
<context>
    <name>Hakkinda</name>
    <message utf8="true">
        <location filename="hakkinda.py" line="78"/>
        <source>%s Hakkında</source>
        <translation>%s Hakkında</translation>
    </message>
    <message>
        <location filename="hakkinda.py" line="79"/>
        <source>%s</source>
        <translation>%s</translation>
    </message>
    <message utf8="true">
        <location filename="hakkinda.py" line="80"/>
        <source>Sürüm %s</source>
        <translation>Sürüm %s</translation>
    </message>
    <message utf8="true">
        <location filename="hakkinda.py" line="81"/>
        <source>Hakkında</source>
        <translation>Hakkında</translation>
    </message>
    <message utf8="true">
        <location filename="hakkinda.py" line="82"/>
        <source>
        &lt;p&gt;Parçala, Hj-Split ile aynı işi yapan dosya parçalama ve birleştirme yazılımıdır.&lt;/p&gt;
        &lt;p&gt;Parçala, büyük boyutlu dosyaları parçalara böldüğü gibi parçalanmış dosyaları birleştirme ve dosyaların doğrulamasını yapabilmektedir.&lt;/p&gt;

        &lt;p&gt;&lt;b&gt;Geliştirici:&lt;/b&gt; Metehan Özbek - &lt;a href=&apos;mailto:metehan@metehan.us&apos;&gt;metehan@metehan.us&lt;/a&gt;&lt;/p&gt;
        &lt;p&gt;&lt;b&gt;Görsel Çalışma:&lt;/b&gt; Yasin Özcan - &lt;a href=&apos;mailto:hamfindik@gmail.com&apos;&gt;hamfindik@gmail.com&lt;/a&gt;&lt;/p&gt;
        &lt;p&gt;&lt;b&gt;Lisans:&lt;/b&gt; GPL v3&lt;/p&gt;
        &lt;p&gt;&lt;/p&gt;</source>
        <translation>
        &lt;p&gt;Parçala, Hj-Split ile aynı işi yapan dosya parçalama ve birleştirme yazılımıdır.&lt;/p&gt;
        &lt;p&gt;Parçala, büyük boyutlu dosyaları parçalara böldüğü gibi parçalanmış dosyaları birleştirme ve dosyaların doğrulamasını yapabilmektedir.&lt;/p&gt;

        &lt;p&gt;&lt;b&gt;Geliştirici:&lt;/b&gt; Metehan Özbek - &lt;a href=&apos;mailto:metehan@metehan.us&apos;&gt;metehan@metehan.us&lt;/a&gt;&lt;/p&gt;
        &lt;p&gt;&lt;b&gt;Görsel Çalışma:&lt;/b&gt; Yasin Özcan - &lt;a href=&apos;mailto:hamfindik@gmail.com&apos;&gt;hamfindik@gmail.com&lt;/a&gt;&lt;/p&gt;
        &lt;p&gt;&lt;b&gt;Lisans:&lt;/b&gt; GPL v3&lt;/p&gt;
        &lt;p&gt;&lt;/p&gt;</translation>
    </message>
</context>
<context>
    <name>ParcalaDialog</name>
    <message utf8="true">
        <location filename="parcala.py" line="43"/>
        <source>Parçala</source>
        <translation>Parçala</translation>
    </message>
    <message>
        <location filename="parcala.py" line="33"/>
        <source>Nereden</source>
        <translation>Nereden</translation>
    </message>
    <message>
        <location filename="parcala.py" line="38"/>
        <source>Nereye</source>
        <translation>Nereye</translation>
    </message>
    <message>
        <location filename="parcala.py" line="56"/>
        <source>Mb</source>
        <translation>Mb</translation>
    </message>
    <message utf8="true">
        <location filename="parcala.py" line="60"/>
        <source>Parçalama Boyutu:</source>
        <translation>Parçalama Boyutu:</translation>
    </message>
    <message>
        <location filename="parcala.py" line="104"/>
        <source>Hata!</source>
        <translation>Hata!</translation>
    </message>
    <message utf8="true">
        <location filename="parcala.py" line="87"/>
        <source>Böyle bir dosya mevcut değil!</source>
        <translation>Böyle bir dosya mevcut değil!</translation>
    </message>
    <message utf8="true">
        <location filename="parcala.py" line="100"/>
        <source>Dosya seçmediniz!</source>
        <translation>Dosya seçmediniz!</translation>
    </message>
    <message>
        <location filename="parcala.py" line="102"/>
        <source>Dosya boyutu girmediniz!</source>
        <translation>Dosya boyutu girmediniz!</translation>
    </message>
    <message utf8="true">
        <location filename="parcala.py" line="104"/>
        <source>Dosya büyüklüğünden fazla değer girdiniz!</source>
        <translation>Dosya büyüklüğünden fazla değer girdiniz!</translation>
    </message>
</context>
<context>
    <name>DogrulaDialog</name>
    <message utf8="true">
        <location filename="dogrula.py" line="39"/>
        <source>Doğrula</source>
        <translation>Doğrula</translation>
    </message>
    <message>
        <location filename="dogrula.py" line="34"/>
        <source>Nereden</source>
        <translation>Nereden</translation>
    </message>
    <message>
        <location filename="dogrula.py" line="74"/>
        <source>Hata!</source>
        <translation>Hata!</translation>
    </message>
    <message utf8="true">
        <location filename="dogrula.py" line="70"/>
        <source>Böyle bir dosya mevcut değil!</source>
        <translation>Böyle bir dosya mevcut değil!</translation>
    </message>
    <message utf8="true">
        <location filename="dogrula.py" line="74"/>
        <source>Dosya seçmediniz!</source>
        <translation>Dosya seçmediniz!</translation>
    </message>
</context>
<context>
    <name>BirlestirDialog</name>
    <message utf8="true">
        <location filename="birlestir.py" line="43"/>
        <source>Birleştir</source>
        <translation>Birleştir</translation>
    </message>
    <message>
        <location filename="birlestir.py" line="33"/>
        <source>Nereden</source>
        <translation>Nereden</translation>
    </message>
    <message>
        <location filename="birlestir.py" line="38"/>
        <source>Nereye</source>
        <translation>Nereye</translation>
    </message>
    <message utf8="true">
        <location filename="birlestir.py" line="67"/>
        <source>001 Dosyası (*.001)</source>
        <translation>001 Dosyası (*.001)</translation>
    </message>
    <message>
        <location filename="birlestir.py" line="83"/>
        <source>Hata!</source>
        <translation>Hata!</translation>
    </message>
    <message utf8="true">
        <location filename="birlestir.py" line="74"/>
        <source>Böyle bir dosya mevcut değil!</source>
        <translation>Böyle bir dosya mevcut değil!</translation>
    </message>
    <message utf8="true">
        <location filename="birlestir.py" line="83"/>
        <source>Dosya seçmediniz!</source>
        <translation>Dosya seçmediniz!</translation>
    </message>
</context>
</TS>


parcala/parcala.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

#Copyright (C) 2011, Metehan Özbek
#
#This program is free software: you can redistribute it and/or modify
#it under the terms of the GNU General Public License as published by
#the Free Software Foundation, either version 3 of the License, or
#(at your option) any later version.
#
#This program is distributed in the hope that it will be useful,
#but WITHOUT ANY WARRANTY; without even the implied warranty of
#MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
#GNU General Public License for more details.
#
#You should have received a copy of the GNU General Public License
#along with this program. If not, see <http://www.gnu.org/licenses/>.

from PyQt4.QtCore import Qt, QSize, QLocale, QTranslator
from PyQt4.QtGui import QWidget, QApplication, QIcon, QPixmap, QGridLayout, QPushButton, QLabel, QSpacerItem, QSizePolicy
from lib import *

import sys
import resource

class Parcala(QWidget):
    def __init__(self):
        QWidget.__init__(self)
        self.resize(350, 225)
        self.setMinimumSize(QSize(350, 225))
        self.setMaximumSize(QSize(400, 250))
        self.setWindowTitle(QApplication.applicationName()+" "+QApplication.applicationVersion())

        icon = QIcon()
        icon.addPixmap(QPixmap(":/resim/parcala.png"))
        self.setWindowIcon(icon)

        self.gridLayout = QGridLayout(self)

        self.parcalaButton = QPushButton(self)
        self.parcalaButton.setMinimumSize(QSize(150, 50))
        self.parcalaButton.setText(self.trUtf8("Parçala"))
        self.parcalaButton.clicked.connect(self.parcala)
        self.gridLayout.addWidget(self.parcalaButton, 3, 0, 1, 1)

        self.birlestirButton = QPushButton(self)
        self.birlestirButton.setMinimumSize(QSize(150, 50))
        self.birlestirButton.setText(self.trUtf8("Birleştir"))
        self.birlestirButton.clicked.connect(self.birlestir)
        self.gridLayout.addWidget(self.birlestirButton, 3, 2, 1, 1)

        self.dogrulaButton = QPushButton(self)
        self.dogrulaButton.setMinimumSize(QSize(150, 50))
        self.dogrulaButton.setText(self.trUtf8("Doğrula"))
        self.dogrulaButton.clicked.connect(self.dogrula)
        self.gridLayout.addWidget(self.dogrulaButton, 7, 0, 1, 1)

        self.hakkindaButton = QPushButton(self)
        self.hakkindaButton.setMinimumSize(QSize(150, 50))
        self.hakkindaButton.setText(self.trUtf8("Hakkında"))
        self.hakkindaButton.clicked.connect(self.hakkinda)
        self.gridLayout.addWidget(self.hakkindaButton, 7, 2, 1, 1)

        self.bilgiLabel = QLabel(self)
        self.bilgiLabel.setText(self.trUtf8(u"%s %s \u00a9 2011 - www.metehan.us"%(QApplication.applicationName(),QApplication.applicationVersion())))
        self.bilgiLabel.setAlignment(Qt.AlignCenter)
        self.gridLayout.addWidget(self.bilgiLabel, 9, 0, 1, 3)
        
        spacerItem = QSpacerItem(40, 20, QSizePolicy.Expanding, QSizePolicy.Minimum)
        self.gridLayout.addItem(spacerItem, 3, 1, 1, 1)
        spacerItem1 = QSpacerItem(0, 20, QSizePolicy.Minimum, QSizePolicy.Expanding)
        self.gridLayout.addItem(spacerItem1, 4, 1, 1, 1)
        spacerItem2 = QSpacerItem(20, 40, QSizePolicy.Minimum, QSizePolicy.Expanding)
        self.gridLayout.addItem(spacerItem2, 2, 1, 1, 1)
        spacerItem3 = QSpacerItem(20, 40, QSizePolicy.Minimum, QSizePolicy.Expanding)
        self.gridLayout.addItem(spacerItem3, 8, 1, 1, 1)


    def hakkinda(self):
        h = Hakkinda(self)
        h.show()

    def dogrula(self):
        d = DogrulaDialog(self)
        d.show()

    def parcala(self):
        d = ParcalaDialog(self)
        d.show()

    def birlestir(self):
        d = BirlestirDialog(self)
        d.show()

def main():
    from os.path import join, abspath, dirname
    mainPath = abspath(dirname(__file__))
    app = QApplication(sys.argv)
    locale = QLocale.system().name()
    translator = QTranslator()
    translator.load(join(mainPath, "language", "%s"%locale))
    app.installTranslator(translator)
    app.setApplicationName(u"Parçala")
    app.setApplicationVersion("1.8")
    
    pencere = Parcala()
    pencere.show()
    sys.exit(app.exec_())

if __name__ == "__main__":
    main()